pisek/jobs/parts/solution.py

```diff
@@ -229,31 +256,99 @@ def _jobs_points(self) -> list[float]:
             )
         )
 
-    def __str__(self) -> str:
-        s = "("
-        previous = list(
-            map(
-                lambda x: x.verdict if x else None,
-                self._job_results(self.previous_jobs),
+    def status(
+        self, all_subtasks: list["SubtaskJobGroup"], verbosity: Optional[int] = None
+    ) -> str:
+        def subtask_name(num: int) -> str:
+            return self._env.config.subtasks[num].name
+
+        def verdict_summary(jobs: list[RunJudge]) -> str:
+            text = ""
+            verdicts = self._judge_verdicts(jobs)
+            for verdict in Verdict:
+                count = verdicts.count(verdict)
+                if count > 0:
+                    text += f"{count}{verdict.mark()}"
+            return text
+
+        def verdict_marks(jobs: list[RunJudge]) -> str:
+            return "".join(job.verdict_mark() for job in jobs)
+
+        verbosity = self._env.verbosity if verbosity is None else verbosity
+        text = ""
+
+        predecessor_summary = verdict_summary(self.previous_jobs)
+        if predecessor_summary:
+            predecessor_summary = f"({predecessor_summary}) "
+
+        if verbosity == 0:
+            text += f"{predecessor_summary}{verdict_marks(self.new_jobs)}"
+
+        elif verbosity == 1:
+            max_sub_name_len = max(
+                len(subtask_name(num)) for num in self._env.config.subtasks
             )
-        )
-        for verdict in Verdict:
-            count = previous.count(verdict)
-            if count > 0:
-                s += f"{count}{verdict.mark()}"
-        s += ") "
-        if s == "() ":
-            s = ""
-
-        for job, result in zip(self.new_jobs, self._job_results(self.new_jobs)):
-            if job.state == State.canceled:
-                s += "-"
-            elif result is None:
-                s += " "
-            else:
-                s += str(result)
-
-        return s
+            max_sub_points_len = max(
+                len(str(sub.points)) for sub in self._env.config.subtasks.values()
+            )
+
+            subtask_name = pad(self.subtask.name + ":", max_sub_name_len + 1)
+            subtask_points = pad_left(format_points(self._points), max_sub_points_len)
+
+            text += right_aligned_text(
+                f"{subtask_name} {subtask_points}p  "
+                f"{predecessor_summary}{verdict_marks(self.new_jobs)} ",
+                f"slowest {self.slowest_time:.2f}s",
+                offset=-2,
+            )
+
+        elif verbosity >= 2:
+            longest_input_name = max(map(lambda j: len(j.input.name), self.new_jobs))
```

Comment:
Nit: Why does this use `map` when `max_sub_name_len` and `max_sub_points_len` use list comprehension? Why is the naming scheme completely different?